lib/rubocop/formatter/simple_text_formatter.rb

```diff
@@ -97,6 +87,55 @@ def message(offense)
         message = offense.corrected? ? green('[Corrected] ') : ''
         message << annotate_message(offense.message)
       end
+
+      # A helper class for building the report summary text.
+      class ReportSummarizer
+        include Colorizable, TextUtil
+
+        def initialize(file_count, offense_count, correction_count, context)
+          @file_count = file_count
+          @offense_count = offense_count
+          @correction_count = correction_count
+
+          @context = context
+        end
+
+        def to_s
+          if @correction_count > 0
+            "#{files} inspected, #{offenses} detected, #{corrections} corrected"
+          else
+            "#{files} inspected, #{offenses} detected"
+          end
+        end
+
+        private
+
+        def files
+          pluralize(@file_count, 'file')
+        end
+
+        def offenses
+          text = pluralize(@offense_count, 'offense', no_for_zero: true)
+          color = @offense_count.zero? ? :green : :red
+
+          colorize(text, color)
+        end
+
+        def corrections
+          text = pluralize(@correction_count, 'offense')
+          color = @correction_count == @offense_count ? :green : :cyan
+
+          colorize(text, color)
+        end
+
+        def options
```

Comment:
*Maybe* I could just pass in the existing `rainbow` instance – that would probably be cleaner.